app.js

```diff
@@ -112,13 +105,13 @@ app.post('/annotate', function(req, res) {
 	let data = ['imageid', 'user', 'annotation', 'feature'].map(attr => req.body[attr]);
 	// if any are not included
 	if (data.some(a => a === undefined)) {
-		res.status(400).send("Error: Invalid data format");
+		res.status(404).send({client: "Error: Invalid data format"});
```

Comment:
I don't this should be a 404. 404's (according to wikipedia) are to indicate that the client was able to communicate with a given server, but the server could not find what was requested. 400's are when the server cannot or will not process the request due to an apparent client error (e.g., malformed request syntax, too large size, invalid request message framing, or deceptive request routing).